docs/src/development/proposals/MEP8/README.md

```diff
@@ -0,0 +1,320 @@
+# Configurable Filesystem layout for Machine Allocation
+
+The current implementation uses a hard coded filesystem layout depending on the specified size and image. This is done in the metal-hammer. This worked well in the past because we had a small amount of sizes and images. But we reached a point where this is to restricted for all use cases we have to fulfill. It also forces us to modify the metal-hammer source code to support a new filesystem layout.
+
+This proposal tries to address this issue by introducing a filesystem layout struct in the metal-api which is then configurable per machine allocation.
+The original behavior of automatic filesystem layout decision must still be present, because there must be no API change for existing API consumers. It should be a additional feature during machine allocation.
+
+## API and behavior
+
+The API will get a new endpoint `filesystemlayouts` (TODO naming) to create/update/delete a set of available `filesystemlayouts`.
+
+A `filesystemlayout` will have the following properties
+
+```go
+// FilesystemLayout to be created on the given machine
+type FilesystemLayout struct {
+  // ID unique layout identifier
+  ID          string
+  // Description is human readable
+  Description string
+  // Filesystems to create on the server
+  Filesystems []Filesystem
+  // Disks to configure in the server with their partitions
+  Disks       []Disk
+  // Raid if not empty, create raid arrays out of the individual disks, to place filesystems onto
+  Raid        []Raid
+  // Constraints which must match to select this Layout
+  Constraints FilesystemLayoutConstraints
+}
+
+type FilesystemLayoutConstraints struct {
+  // Sizes defines the list of sizes this layout applies to
+  Sizes []string
+  // Images defines a list of image glob patterns this layout should apply
+  // the most specific combination of sizes and images will be picked fo a allocation
+  // If prefixed with "!" this image glob is not allowed
+  Images []string
```

Comment:
The whole selection strategy is the most difficult part here. I was also thinking about supporting semantic versioning for images. The actual mechanism to the create the fstab actually relies on a implementation of the fstab creation in the image. With this proposal, this approach wont work anymore because doing this in a shell script is a hassle. I would therefore implement this in the metal-hammer. Then its obviously that older images wont work anymore. But OTOH, metal-hammer is deployed partition wide which means that this new functionality is there already.

Reply:
Clarified the Constraints a bit, changed from glob to semver for images, please check again if this is now more appropriate

Reply:
Changed to semver image constraints incl. validation if not ambigous